app/Http/Controllers/MainController.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Post;
+use Illuminate\Http\Request;
+use Aws\S3\S3Client;
+use Illuminate\Support\Str;
+
+class MainController extends Controller
+{
+    protected S3Client $s3;
+
+    public function __construct(S3Client $s3) {
+        $this->s3 = $s3;
+    }
+
+    public function showMain()
+    {
+        $posts = Post::latest()->get();
+        return view('main', ['posts' => $posts]);
+    }
+
+    public function uploadPost(Request $request) {
+        $request->validate([
+            'image' => 'required|file|mimes:jpeg,jpg|max:1024', // max 1MB
+            'caption' => 'required'
+        ]);
+
+        // upload image to s3, here we are using s3 client from AWS SDK PHP
+        // instead storage library provided by Laravel, the reason is that
+        // the Laravel storage library doesn't throw any exception when error
```

Comment:
To show an error, need to change throw to true in the /config/filesystems.php file
```
        's3' => [
            'driver' => 's3',
            'key' => env('AWS_ACCESS_KEY_ID'),
            'secret' => env('AWS_SECRET_ACCESS_KEY'),
            'region' => env('AWS_DEFAULT_REGION'),
            'bucket' => env('AWS_BUCKET'),
            'url' => env('AWS_URL'),
            'endpoint' => env('AWS_ENDPOINT'),
            'use_path_style_endpoint' => env('AWS_USE_PATH_STYLE_ENDPOINT', false),
            'throw' => false,
        ],
```

Reply:
Hoo.., I just know about this, @fatkulnurk. Thanks a lot. 📝

Reply:
I have implemented your suggestion on https://github.com/farulivan/jepret/pull/13/commits/0780a810e60e8c1b0b1fde8f0ea43ec6d98106a6, @fatkulnurk. 

Please kindly recheck.